app/models/katello/concerns/content_facet_host_extensions.rb

```diff
@@ -154,15 +155,25 @@ def find_by_repository_name(_key, operator, value)
           end
         end
 
-        def in_content_view_environment(content_view: nil, lifecycle_environment: nil)
-          relation = self.joins(:content_facet)
-          relation = relation.where("#{::Katello::Host::ContentFacet.table_name}.content_view_id" => content_view) if content_view
-          relation = relation.where("#{::Katello::Host::ContentFacet.table_name}.lifecycle_environment_id" => lifecycle_environment) if lifecycle_environment
+        def in_content_view_environments(content_views: nil, lifecycle_environments: nil)
+          relation = self.joins(:content_facet => {:content_view_environment_content_facets => :content_view_environment })
+          relation = relation.where("#{::Katello::ContentViewEnvironment.table_name}.content_view_id" => content_views) if content_views
+          relation = relation.where("#{::Katello::ContentViewEnvironment.table_name}.environment_id" => lifecycle_environments) if lifecycle_environments
           relation
         end
 
+        def in_environments(lifecycle_environments)
+          in_content_view_environments(:lifecycle_environments => lifecycle_environments)
+        end
+
+        def in_content_view_environment(content_view: nil, lifecycle_environment: nil)
+          Rails.logger.warn "DEPRECATION WARNING: in_content_view_environment is deprecated and will be removed in Katello 4.8.  Please use in_content_view_environments instead."
+          in_content_view_environments(:content_views => content_view, :lifecycle_environments => lifecycle_environment)
+        end
+
         def in_environment(lifecycle_environment)
-          in_content_view_environment(:lifecycle_environment => lifecycle_environment)
+          Rails.logger.warn "DEPRECATION WARNING: in_environment is deprecated and will be removed in Katello 4.9.  Please use in_environments instead."
```

Comment:
Do you think it's weird to have deprecation warnings for non-user facing things? My thinking is that it's something outside of their control, so they won't be able to avoid the warnings.

Reply:
Nah, we have a ton of those in the kings already 🤷🏼‍♂️